File: flutter_project/lib/models/student.dart
import 'package:hive/hive.dart';
part 'student.g.dart';

@HiveType(typeId: 0)
class Student extends HiveObject {
  @HiveField(0)
  String? name;
  @HiveField(1)
  String? id;
  @HiveField(2)
  String? email;
  @HiveField(3)
  String? password;
  @HiveField(4)
  String? gender;
  @HiveField(5)
  String? level;
  @HiveField(6)
  String? image;
  @HiveField(7)
  List<int>? favStores;

  Student(
      {this.name,
      this.id,
      this.email,
      this.password,
      this.gender,
      this.level,
      this.image,
      this.favStores});

  Student.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    id = json['id'];
    email = json['email'];
    password = json['password'];
    gender = json['gender'];
    level = json['level'];
    image = json['image'];
    favStores = List<int>.from(json['favStores'] ?? []);
  }
  @override
  String toString() {
    return 'Student{name: $name, email: $email, id: $id, password: $password, gender: $gender, level: $level, image: $image, favStores: $favStores}';
  }
}

File: springboot/api/src/main/java/com/store/api/entity/Student.java
package com.store.api.entity;

import jakarta.persistence.*;
import java.util.List;

@Entity
public class Student {
    @Id
    private String id;
    private String name;
    private String email;
    private String password;
    private String gender;
    private String level;
    private String image;

    @ElementCollection
    private List<Integer> favStores;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public List<Integer> getFavStores() {
        return favStores;
    }

    public void setFavStores(List<Integer> favStores) {
        this.favStores = favStores;
    }
}
